File: two-sum.java
import java.util.*;
class Main{
  public static void main(String []args){
    int []nums={2,7,11,15};
    int target=9;
    int []ar=findLocations(nums, target);
    for(int i:ar){
      System.out.println(i);
    }
  }
  public static int[] findLocations(int []nums, int target){
    if(nums.length<2){
      return new int[] {-1, -1};
    }
    HashMap<Integer, Integer> map=new HashMap<>();
    int []result=new int [2];
    for(int i=0;i<nums.length;i++){
      int temp=target-nums[i];
      if(!map.containsKey(temp)){
        map.put(nums[i], i);
      } else {
          result[0]= map.get(temp);
          result[1]=i;
      }
    }
    return result;
  }
}

File: TwoSum.cs
//Time: O(n)
//Space: O(n)
//Leetcode: https://leetcode.com/submissions/detail/575536540/

using System;

namespace twoSum
{   

    public int[] TwoSum(int[] nums, int target) {
        
        int[] result = new int[2];
        Dictionary<int, int> dict = new Dictionary<int, int>();
        
        for(int i =0; i< nums.Length; i++) {
            int complement = target - nums[i];
            if(dict.ContainsKey(complement)) {
                result[1] = i;
                result[0] = dict[complement];
            } else {
                if(! dict.ContainsKey(nums[i]))
                    dict.Add(nums[i], i);
            }            
        }
        return result;
    }

    class Program
    {
        static void Main(string[] args)
        {
            
            Console.WriteLine("Hello World!");
        }
    }
}
